### PR TITLE
[EOC-509] Remove data-icon form icons components and add aria label

### PR DESCRIPTION
**Pull request by alemikolo**
I added aria-label and title props to the Icons components. But in almost all cases passing those props to the icon is unnecessary because:
1. We use icons, especially in buttons or links that already have their titles attributes which are used by screen readers. 
2. We use icons only for visual purposes next to titles, headers or labels so they have no functional significance.

I passed labels only in places where icon displays additional info which isn't displayed in another way eg. `lock list icon`.

Some buttons had no title attribute so I added it.

### CODE REVIEW

File: client/src/assets/images/icons/index.js [hunk 1]
```diff
@@ -52,7 +52,7 @@ export const PlusIcon = () => (
 );
 
 export const RegularStarIcon = () => (
-  <Icon data-icon="star" viewBox="0 0 576 512">
+  <Icon aria-label="fot favourite" viewBox="0 0 576 512">
```
Comment (adamklepacz) [inline client/src/assets/images/icons/index.js]: fot favourite?

File: client/src/assets/images/icons/index.js [hunk 2]
```diff
@@ -97,7 +97,7 @@ export const UserIcon = () => (
 );
 
 export const DotsIcon = () => (
-  <Icon data-icon="ellipsis-h" viewBox="0 0 512 512">
+  <Icon aria-label="ellipsis" rviewBox="0 0 512 512">
```
Comment (adamklepacz) [inline client/src/assets/images/icons/index.js]: `rviewBox`

typo

File: client/src/assets/images/icons/index.js [hunk 3]
```diff
@@ -124,7 +124,7 @@ export const LogoutIcon = () => (
 );
 
 export const ChevronRight = () => (
-  <Icon data-icon="chevron-right" viewBox="0 0 320 512">
+  <Icon viewBox="0 0 320 512">
```
Comment (adamklepacz) [inline client/src/assets/images/icons/index.js]: no aria-label?

File: client/src/assets/images/icons/index.js [hunk 4]
```diff
@@ -178,23 +178,21 @@ export const ListViewIcon = () => (
...
       fill="#000"
     />
-  </svg>
+  </Icon>
 );
 
 export const CheckIcon = () => (
-  <Icon data-icon="check" viewBox="0 0 512 512">
+  <Icon aria-label="ok" viewBox="0 0 512 512">
```
Comment (adamklepacz) [inline client/src/assets/images/icons/index.js]: maybe `check mark` would be more reasonable for people using aria labels?

File: client/src/assets/images/icons/index.js [hunk 5]
```diff
@@ -212,25 +210,16 @@ export const ErrorIcon = () => (
 );
 
 export const AppIcon = () => (
-  <Icon data-icon="coffee" viewBox="0 0 640 512">
+  <Icon aria-label="eoc" viewBox="0 0 640 512">
```
Comment (adamklepacz) [inline client/src/assets/images/icons/index.js]: maybe 'aplication logo' will be more reasonable for people using aria labels?

File: client/src/assets/images/icons/index.js [hunk 6]
```diff
@@ -25,7 +25,7 @@ export const CohortIcon = () => (
 );
 
 export const ListIcon = () => (
-  <Icon data-icon="clipboard-list" viewBox="0 0 384 512">
+  <Icon aria-label="list" viewBox="0 0 384 512">
```
Comment (Ciunkos) [inline client/src/assets/images/icons/index.js]: Please investigate https://stackoverflow.com/questions/22039910/what-is-aria-label-and-how-should-i-use-it

File: client/src/app/components/Layout/Layout.jsx [hunk 7]
```diff
@@ -162,8 +162,8 @@ export class Layout extends PureComponent {
               title={formatMessage({
                 id:
                   viewType === ViewType.LIST
-                    ? 'app.layout.list-view'
-                    : 'app.layout.tile-view'
+                    ? 'app.layout.tile-view'
```
Comment (przemyslawzalewski) [inline client/src/app/components/Layout/Layout.jsx]: ??

Comment (alemikolo) [reply client/src/app/components/Layout/Layout.jsx]: There was a bug. Message-ids were reversed. Now it's ok. If there is a list view we want to change it to a tile view. So button title should be: 'change to tile view'

Comment (Ciunkos) [reply client/src/app/components/Layout/Layout.jsx]: Consider renaming locale key then, i.e. "app.layout.switch-to-tile-view".

**Pull request merged**